lib/matplotlib/cm.py

```diff
@@ -487,9 +487,12 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                 else:
                     raise ValueError("Third dimension must be 3 or 4")
                 if xx.dtype.kind == 'f':
-                    if norm and (xx.max() > 1 or xx.min() < 0):
+                    if norm and (
+                            np.any(np.isnan(xx)) or (xx.max() > 1 or xx.min() < 0)
+                    ):
```

Comment:
Rewrite as
`if not (xx.min() >= 0 and xx.max() <= 1)` (with a comment explaining that this form also excludes nans) to save a tiny bit of performance?